src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  // Tablica, która będzie przechowywać historię stanów
+  const stateHistory = [];
+
+  // Zaczynamy od klonowania początkowego stanu
+  let currentState = { ...state };
+
+  // Iterujemy po każdej akcji
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        // Tworzymy nowy pusty obiekt stanu
+        currentState = {};
+        break;
+
+      case 'addProperties':
+        // Dodajemy nowe właściwości do sklonowanego stanu
+        currentState = { ...currentState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        // Klonujemy stan i usuwamy wskazane klucze
+        currentState = { ...currentState };
```

Comment:
[CODE STYLE] - There is no need to create a new copy of `currentState` on this line since you are not mutating the original `state` object here. You can directly delete the properties from `currentState`.